contracts/RMRK/equippable/RMRKEquippable.sol

```diff
@@ -196,32 +196,26 @@ contract RMRKEquippable is
     /**
      * @notice Used to add a resource entry.
      * @dev This internal function warrants custom access control to be implemented when used.
-     * @dev `ExtendedResource` consists of the following parameters:
-     *  [
-     *      resourceId,
-     *      childResourceId,
-     *      childId,
-     *      childEquippableAddress
-     *  ]
-     * @param resource An `ExtendedResource` struct containing the components of a resource we are adding
      * @param fixedPartIds An array of IDs of fixed parts to be included in the resource
      * @param slotPartIds An array of IDs of slot parts to be included in the resource
      */
     function _addResourceEntry(
-        ExtendedResource memory resource,
-        uint64[] calldata fixedPartIds,
-        uint64[] calldata slotPartIds
+        uint64 id,
+        uint64 equippableGroupId,
+        address baseAddress,
+        string memory metadataURI,
+        uint64[] memory fixedPartIds,
+        uint64[] memory slotPartIds
     ) internal virtual {
-        uint64 id = resource.id;
-        _addResourceEntry(id, resource.metadataURI);
+        _addResourceEntry(id, metadataURI);
 
         if (
-            resource.baseAddress == address(0) &&
+            baseAddress == address(0) &&
             (fixedPartIds.length != 0 || slotPartIds.length != 0)
         ) revert RMRKBaseRequiredForParts();
```

Comment:
Having a Base is a requirement for each Equippable?

Reply:
So having these fields you have to define for each new resource where it can be equipped. 
Have I got that correctly?

Reply:
If you're setting either fixed or slot parts, there has to be a base. Otherwise it can be address 0.
The fixed parts, are the parts this resource will use. The slot parts, are the ones it can receive. (Both should be defined in the same base).
If you want to add a single resource with nothing from the base nor expected to equip something, you can set just the id and metadataURI, the rest would be empty.

Reply:
I see, that sounds good, thanks.